### ISSUE TITLE
GtiHub module (to access GH API from GH Action)

### ISSUE THREAD

**Issue by applejag** [risor-io/risor#124]
Using Risor could be a nice contestant to using Bash in GH Actions.

Not talking about releasing to GitHub, but instead to use the go-github package: https://pkg.go.dev/github.com/google/go-github/v57/github

It's such a huge package that not everything needs to be there. But common tasks used in GH actions, like:

- Automatically use GITHUB_TOKEN secret for credentials

- PRs:
  - Create/find/update comments
  - Create/find/update reviews

- GH Action cache:
  - Upload/download/delete dependency cache (Like the cache cache library https://github.com/actions/toolkit/tree/main/packages/cache)
  - Upload/download/delete tool cache (Like the tool-cache library https://github.com/actions/toolkit/tree/main/packages/tool-cache)
  - Helper function that does a "create or update comment" by using `<!-- risor-id: 1234 -->` string in the markdown comment so it can find it again on next call.

- Read GH Action/workflow inputs

- GH Action commands helpers (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions)
  - Logging helpers (info/err/notice)
  - Set env vars
  - Set output values

- Upload/download artifacts
 
All of this is already doable, but having them wrapped in easy-to-use first-class module would be nice.

Alternatively, this could also be a third-party module you'll have to `import`. What's the decision like for what goes into the "standard library" and what not?

**Comment by myzie** [comment]
Yeah, something along these lines has crossed my mind too. I'm 100% onboard with some form of wrapping the go-github client. And agree that use in GH actions would make a lot of sense.

**Comment by applejag** [comment]
I would really like to attempt adding this, but I'm off on holiday until next year.

OK if I call dibs on it? :)

**Comment by myzie** [comment]
Of course 😄 you’ve got dibs!

**Comment by myzie** [comment]
Hi @applejag, now that the github actions module is in place, what do you think this issue should cover?

**Comment by applejag** [comment]
I'd like to try completing basically everything on that list on the issue. My intention was to do it iteratively, to not have a too huge of a PR.

Time has been limited recently, so I lost my flow.

But I still want to continue on this.

**Comment by applejag** [comment]
Btw I'll not be implementing this. You can close it if you want @myzie

**Comment by myzie** [comment]
Roger that @applejag, I'll just unassign you from the ticket.

You using Risor at all these days?

**Comment by applejag** [comment]
> Roger that @applejag, I'll just unassign you from the ticket.
> 
> You using Risor at all these days?

No, sorry I'm not. I tried using it at my company but there were too many sharp edges and so we changed to just writing our scripts plain Go instead. The setup-go GitHub action is quite slow, but the biggest dealbreaker for us was the lack of unit tests. For us a script is either small enough to be written in bash, or if it gets just slightly complicated we move over to Go. And then we just skip all the languages that fit "in between" so to speak, which includes languages like JavaScript, Python, and Risor.

I'm actually currently eyeing MoonBit if that could be a good contender. Or perhaps going full out dagger.io for a more "declarative approach". But sadly Risor felt a bit too much like paying the early adopter tax.

I also tried using it in some hobby projects but the sharp edges made me move away from it. Some of those sharp edges were the error handling I remember. But it was a while ago so I don't remember all the reasons.

**Comment by rubiojr** [comment]
Happy to pick it 🆙

**Comment by myzie** [comment]
First pass on the Github module is now available:
https://github.com/risor-io/risor/pull/403

Further additions can be done in the future as needed.
